web/templates/stream_sidebar_row.hbs

```diff
@@ -10,11 +10,13 @@
 
             <a href="{{url}}" title="{{name}}" class="stream-name">{{name}}</a>
 
+            {{#if can_post_messages}}
             <div class="left-sidebar-controls">
                 <div class="channel-new-topic-button tippy-zulip-tooltip hidden-for-spectators" data-tippy-content="{{t 'New topic'}}" data-stream-id="{{id}}">
                     <i class="channel-new-topic-icon zulip-icon zulip-icon-square-plus" aria-hidden="true"></i>
                 </div>
             </div>
+            {{/if}}
```

Comment:
@karlstolley do we want the if statement inside the `left-sidebar-controls` div or outside it?

Reply:
Probably inside. It possible that other controls might go there in the future, and it would be cleaner to have this targeted just to the new topic button.

@pratik-pc, double-check that there's no layout side-effects with the more targeted approach. The grid should collapse to zero, but just to be sure.